### DISCUSSION TITLE
help for docker-compose with valhalla.json config

### DISCUSSION THREAD

**Discussion by mertbakir** [valhalla/valhalla#3647] [Q&A] [answered]
Hi, I used this image to start the valhalla container: https://github.com/gis-ops/docker-valhalla and got an error.

```
terminate called after throwing an instance of 'boost::wrapexcept<boost::property_tree::ptree_bad_path>'
  what():  No such node (service_limits.auto.max_matrix_location_pairs)
/valhalla/scripts/run.sh: line 9:    96 Aborted                 sudo -E $1
```

I think it's because of the configuration file. I had `valhalla.json` config before but it doesn't work now. I looked for a template but couldn't find. Can you point me a template configuration?

Also, is there specific path that I should mount the configuration or is it ok to place it under `custom_files`

**Comment by nilsnolde** [comment, marked as answer]
that generally happens very frequently: we add some mandatory iterms to the configuration which is not forward-compatible with older config JSONs. your best bet is to produce a new config JSON after every `docker pull` (or only if there's an error). if you have only defaults anyways, simply remove the file and restart the container. if you don't then you'll need to either patch the new valhalla json the container created or create a new one before you start the container by running smth like `docker run --rm -v $PWD/custom_files:/custom_files gisops/valhalla:latest valhalla_build_config (--help/--<options>) > /custom_files/valhalla.json` (untested)

script is here: https://github.com/valhalla/valhalla/blob/master/scripts/valhalla_build_config

the config _must_ be in the folder you mount inside container, otherwise the container won't see it.

**Reply by mertbakir** [reply]
Thanks!
